pkgs/development/compilers/ccl/default.nix

```diff
@@ -1,23 +1,44 @@
 { stdenv, fetchsvn, gcc, glibc, m4, coreutils }:
 
-/* TODO: there are also MacOS, FreeBSD and Windows versions */
-assert stdenv.system == "x86_64-linux" || stdenv.system == "i686-linux";
-
+let
+  options = rec {
+    x86_64-linux = {
+      arch = "linuxx86";
+      sha256 = "04p77n18cw0bc8i66mp2vfrhlliahrx66lm004a3nw3h0mdk0gd8";
+      runtime = "lx86cl64";
+      kernel = "linuxx8664";
+    };
+    i686-linux = {
+      arch = "linuxx86";
+      sha256 = x86_64-linux.sha256;
+      runtime = "lx86cl";
+      kernel = "linuxx8632";
+    };
+    armv7l-linux = {
+      arch = "linuxarm";
+      sha256 = "0xg9p1q1fpgyfhwjk2hh24vqzddzx5zqff04lycf0vml5qw1gnkv";
+      runtime = "armcl";
+      kernel = "linuxarm";
+    };
+    armv6l-linux = armv7l-linux;
+  };
+  cfg = options.${stdenv.system};
```

Comment:
i added the assert back for now

@vcunat it doesnt seem to me that "assert options ? stdenv.system;" would do the desired thing because both things are always true, or is my understanding wrong?  i tried assert cfg but the error happes already in the "cfg = options.${stdenv.system};" line.  is there a way for cfg to be nil if options doesnt have that slot?